sgl-kernel/csrc/cpu/moe.cpp

```diff
@@ -1171,12 +1176,18 @@ at::Tensor shared_expert_cpu(
   //   3. Aq_tmp : [M, K] or [M, N]
   //   4. As_tmp : [M]
   //
+  // for fp8 w8a16:
+  //   3. Bdq_tmp : [2N, K] or [K, N]
+  //   4. Bs_tmp :  [2N] o [K]
   int num_threads = at::get_num_threads();
   int64_t buffer_size_nbytes = M * N * 2 + num_threads * 2 * BLOCK_M * BLOCK_N * sizeof(float);
 
   if (use_int8_w8a8) {
     buffer_size_nbytes += std::max(M * K, M * N) + M * sizeof(float);
   }
+  if (use_fp8_w8a16) {
+    buffer_size_nbytes += std::max(2 * N * K, K * N) * 2 + std::max(2 * N, K) * sizeof(float);
+  }
```

Comment:
if you follow https://github.com/mingfeima/sglang/pull/47 then we don't need extra memory for Bdq_tmp and Bs_tmp

so in `fp8_scaled_mm_kernel_impl` from the above PR, it only uses Btmp which is [BLOCK_N, BLOCK_K] bfloat16 to store the bfloat16 values for the B block. And then do brgemm on bf16